tests/tests_full_integration_dyndns.yml

```
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: MIT
+---
+
+# To run this test, AD (ad1) needs to be present in the inventory
+# AD should be pre-configured as a dns for the client machine.
+# AD's DNS service should be configured with both forward and reverse zones
+# and the DNS records from any previous runs removed.
+# Note that both for client and ad the ansible_host should
+# contain the ip4 address to make this work properly.
+# There is expectation that both client and ad are on the
+# same network with mask 255.255.255.0.
+
+# Example inventory:
+# [client]
+# client1 ansible_host=<CLIENT IP>\
+#  ad_integration_password=Secret123 ad_integration_realm="domain.com" \
+#  ad_dyndns_update=true
+# [ad]
+# ad1 ansible_host=<AD IP> ansible_connection=winrm ansible_password=Secret123\
+#  ansible_port=5986 ansible_user=Administrator \
+#  ansible_winrm_server_cert_validation=ignore
+
+- name: Set variables
+  hosts: client
+  tasks:
+    - name: Set ad_integration_realm if it is not defined or is empty
+      set_fact:
+        ad_integration_realm: domain.com
+      when: ad_integration_realm is not defined or not ad_integration_realm
+
+- name: Configure DynDNS on AD
+  hosts: ad
+  gather_facts: true
+  tasks:
+    - name: Get AD address
+      set_fact:
+        ad_address: "{{ hostvars[inventory_hostname].ansible_host }}"
+    - name: Get network for zone
+      set_fact:
+        network_ad: >-
+          "{{ ad_address.split('.')[-2::-1] | join('.') }}.in-addr.arpa"
+    - name: show network
+      debug:
+        msg: "AD network: {{ network_ad }}"
+    - name: Create reverse zone
+      win_command: dnscmd.exe /zoneadd {{ network_ad }} /primary
+      register: res
+      failed_when:
+      - 'res.rc != 0 and "DNS_ERROR_ZONE_ALREADY_EXISTS" not in res.stdout'
+    - name: Allow updates in the zone
+      win_command: dnscmd.exe /config {{ hostvars[groups['client'][0]].
+        ad_integration_realm }} /allowupdate 1
+    - name: Allow updates in the reverse zone
+      win_command: dnscmd.exe /config {{ network_ad }} /allowupdate 1
+    - name: Disable dns forwarders
+      win_command: dnscmd.exe /config /norecursion 1
+
+- name: Ensure that the role configures dynamic dns
+  hosts: client
+  tasks:
+    - name: Run the integration
+      block:
```

```suggestion
      when: "'ad' in groups and groups['ad']"
      block:
```
The block/rescue/always should come after any `when`, `vars`, etc.